File: ruby/pangram/pangram.rb
class Pangram

  ALPHABET = 'abcdefghijklmnopqrstuvwxyz'

  def self.is_pangram?(str)
    ALPHABET == str.downcase.scan(/[a-z]/).uniq.sort.join
  end
end

module BookKeeping
  VERSION = 2
end

File: ecmascript/pangram/pangram.js
const ALPHABET = 'abcdefghijklmnopqrstuvwxyz';

class Pangram {
  constructor(input) {
    this.input = input;
  }

  isPangram() {
    return this.letters() === ALPHABET;
  }

  letters() {
    const letters = this.input.toLowerCase().replace(/[^a-z]/gi, '').split('');
    return [...new Set(letters)].sort().join('');
  }
}


export default Pangram;
